services/subscriber/service.go

```diff
@@ -219,19 +182,82 @@ func (s *Service) Points() chan<- *coordinator.WritePointsRequest {
 }
 
 // read points off chan and write them
-func (s *Service) writePoints() {
+func (s *Service) run() {
 	defer s.wg.Done()
-	for p := range s.points {
-		for se, sub := range s.subs {
-			if p.Database == se.db && p.RetentionPolicy == se.rp {
-				err := sub.WritePoints(p)
+	subs := make(map[subEntry]PointsWriter)
+	// Perform initial update
+	s.updateSubs(subs)
+	for {
+		select {
+		case <-s.update:
+			err := s.updateSubs(subs)
+			if err != nil {
+				s.Logger.Println("failed to update subscriptions:", err)
+			}
+		case p, ok := <-s.points:
+			if !ok {
+				return
+			}
+			for se, sub := range subs {
+				if p.Database == se.db && p.RetentionPolicy == se.rp {
+					err := sub.WritePoints(p)
```

Comment:
Each HTTP subscription creates a new BatchPoints, then each `Client.Write()` call creates a new `bytes.Buffer` where it then encodes all of the points to line protocol, writing that buffer to the destined remote URL. That buffer is then garbage collected, and the next HTTP subscription will have to redo all of that work.

If it's not common to have more than one subscription per DB/RP, then trying to prevent that waste wouldn't do anything.